RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -925,6 +992,68 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent, const edm::EventSetup& iSetu
      }
    }
 
+
+  // Fill ME0 info
+
+   if (doME0_){
+     if ( me0HitHandle_.isValid() )
+       {
+   	 for(  const auto& chamber : info.chambers )
+   	   {
+   	     if (chamber.id.subdetId() != MuonSubdetId::ME0 ) continue; // Consider ME0 chambers only
+	  
+   	     reco::MuonChamberMatch matchedChamber;
+
+   	     //LocalError localError = chamber.tState.localError().positionError();
+   	     const auto& localError = chamber.tState.localError().positionError();
+   	     matchedChamber.x = chamber.tState.localPosition().x();
+   	     matchedChamber.y = chamber.tState.localPosition().y();
+   	     matchedChamber.xErr = sqrt( localError.xx() );
+   	     matchedChamber.yErr = sqrt( localError.yy() );
+
+   	     matchedChamber.dXdZ = chamber.tState.localDirection().z()!=0?chamber.tState.localDirection().x()/chamber.tState.localDirection().z():9999;
+   	     matchedChamber.dYdZ = chamber.tState.localDirection().z()!=0?chamber.tState.localDirection().y()/chamber.tState.localDirection().z():9999;
+   	     // DANGEROUS - compiler cannot guaranty parameters ordering
+   	     AlgebraicSymMatrix55 trajectoryCovMatrix = chamber.tState.localError().matrix();
+   	     matchedChamber.dXdZErr = trajectoryCovMatrix(1,1)>0?sqrt(trajectoryCovMatrix(1,1)):0;
+   	     matchedChamber.dYdZErr = trajectoryCovMatrix(2,2)>0?sqrt(trajectoryCovMatrix(2,2)):0;
+
+   	     matchedChamber.edgeX = chamber.localDistanceX;
+   	     matchedChamber.edgeY = chamber.localDistanceY;
+
+   	     matchedChamber.id = chamber.id;
+   	     // for ( ME0SegmentCollection::const_iterator me0Segment = me0Segments->begin();
+   	     // 	me0Segment != me0Segments->end(); ++me0Segment )
+   	     for ( const auto& me0RecHit : *me0HitHandle_ )
+   	       {
+   		 reco::MuonSegmentMatch me0HitMatch;
+
+   		 //ME0DetId SegIdPreCompare (me0RecHit.rawId());
+   		 ME0DetId SegIdForCompare (me0RecHit.me0DetId().region(),
+   					   me0RecHit.me0DetId().layer(),
+   					   me0RecHit.me0DetId().chamber(),
+   					   0);
+   		 if ( SegIdForCompare != chamber.id.rawId() ) continue;
+
+   		 me0HitMatch.x = me0RecHit.localPosition().x();
+   		 me0HitMatch.mask = 0;
+
+   		 const double AbsDx = std::abs(me0RecHit.localPosition().x()-chamber.tState.localPosition().x());
+   		 const double AbsDy = std::abs(me0RecHit.localPosition().y()-chamber.tState.localPosition().y());
+   		 // std::cout<<"AbsDx = "<<AbsDx<<", PullX = "<<sqrt(localError.xx() + me0RecHit.localPositionError().xx())<<std::endl;
+   		 // std::cout<<"AbsDy = "<<AbsDy<<", PullY = "<<sqrt(localError.yy() + me0RecHit.localPositionError().yy())<<std::endl;
+
+   		 //These matches are hardcoded, is that okay?
+   		 if( (AbsDx <= 3 or AbsDx/sqrt(localError.xx() + me0RecHit.localPositionError().xx()) <= 4) and 
+   		     (AbsDy <= 9999. or AbsDy/sqrt(localError.yy() + me0RecHit.localPositionError().yy()) <= 9999.)
```

Comment:
Those local errors should be > 0, at least with the same likelihood as with other segments used elsewhere in the MuonIdProducer.